docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+  - Example of event emission:
+    ```cadence
+    emit Transfer(from: sender, to: recipient, amount: 100.0)
+    ```
+
+Events are an essential part of the Flow blockchain ecosystem, enabling communication between the blockchain and external services while preserving security and transparency.
+
+---
+
+This section outlines the core constructs of **contracts**, **interfaces**, and **transactions** in Cadence. These features form the backbone of decentralized applications on Flow, providing developers with powerful tools to create secure, scalable, and flexible smart contracts and user interactions.
+
+## VIII. Memory Model and Storage
+
+The **memory model and storage system** in Cadence are designed to securely and efficiently manage data and resources in a decentralized context. This section explains how data is organized in Flow accounts, the scope of variables, and how resources and data persist across transactions.
+
+---
+
+### 1. Account Storage
+
+In Cadence, data and resources are stored within Flow accounts using a well-defined structure to control access and ownership. Every account has three distinct **storage paths**—`storage`, `private`, and `public`—which dictate who can access stored data.
+
+- **Explanation of Storage Paths**:
+  - **`storage`**: This is the primary location for storing account-specific data and resources. Data in `storage` is only accessible by the account owner. It is the most secure and private storage path, where valuable assets such as NFTs, tokens, or sensitive contract state can be stored.
+    - Example: `/storage/myToken`.
+  
+  - **`private`**: Data stored in `private` paths can only be accessed by the account itself or by authorized parties through **capabilities**. The `private` path is used for data that should not be exposed to the public but may be shared with specific contracts or users via capabilities.
+    - Example: `/private/mySecretData`.
+
+  - **`public`**: Data stored in `public` paths is accessible to anyone. Public storage is typically used for exposing data that other accounts or contracts may need to interact with, such as balances or public-facing information. Public paths do not expose the actual resource but rather provide references or capabilities to interact with the data.
+    - Example: `/public/tokenBalance`.
+
+- **Rules for Storing Resources and Data**:
+  - Resources and data must be stored in well-defined paths under one of the storage categories (`storage`, `private`, or `public`). Resources are moved between accounts using explicit transactions and must be stored in the correct paths to ensure security and proper access control.
+  - Only the account owner can directly manipulate their `storage` paths. Public and private paths must use **capabilities** to grant or restrict access to external
```

Comment:
Maybe mention here that Cadence uses lexical scoping (as opposed to e.g. dynamic scoping)